### PR TITLE
Fix AC-146 and AC-147

### PR DESCRIPTION
**Pull request by AvijitGhosh82**
Patient list loading crashed if the birthday was null (AC-147)
The app did not connect to server if user did not explicitly mention http or https (AC-146)
Both fixed.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/LoginActivity.java [hunk 1]
```diff
@@ -285,6 +289,15 @@ public void setUrl(String url) {
         }
     }
 
+    public String ensure_has_protocol(String a_url)
```
Comment (rkorytkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/LoginActivity.java]: Please follow java conventions when naming methods. Do not use _.

File: openmrs-client/src/main/java/org/openmrs/mobile/net/AuthorizationManager.java [hunk 2]
```diff
@@ -33,9 +38,17 @@ public void login(LoginListener listener) {
         String url = listener.getServerURL() + LOGIN_END_URL;
         mLogger.i(SENDING_REQUEST + url);
 
+        Response.ErrorListener errorListener = new Response.ErrorListener() {
+
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                error.printStackTrace();
```
Comment (rkorytkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/net/AuthorizationManager.java]: Handle it properly. Use http://developer.android.com/reference/android/util/Log.html

What error message is displayed to the user?

Comment (AvijitGhosh82) [inline openmrs-client/src/main/java/org/openmrs/mobile/net/AuthorizationManager.java]: This particular code is for fixing AC-137 (Work in progress) and unrelated to the NPE. I've added the log commit though, but I'll examine the actual log later as I get time.

**Comment by AvijitGhosh82** [comment]
Ping @rkorytkowski

**Comment by rkorytkowski** [comment]
Could you please try sending one PR per issue (especially do not include any code related to other issues without mentioning the issue id in the commit message)? If it is a quick fix, you could also push without a PR and ask for a post commit review by adding the commit URL in the comment on the issue.

**Pull request merged**